lib/azure/armrest/armrest_service.rb

```diff
@@ -137,11 +137,15 @@ def self.fetch_subscription_id(config)
           :authorization => config.token
         )
 
-        subscriptions = JSON.parse(response)["value"]
+        hash = JSON.parse(response)["value"].first
```

Comment:
Shouldn't you still need to find the first enabled subscription? Use and warn the first subscription only if no enabled subscription is found. Raise error if no subscription is found at all.